packages/nestjs/src/setup.ts

```diff
@@ -78,6 +78,30 @@ class SentryGlobalFilter extends BaseExceptionFilter {
 Catch()(SentryGlobalFilter);
 export { SentryGlobalFilter };
 
+/**
+ * Global filter to handle exceptions and report them to Sentry.
+ *
+ * The BaseExceptionFilter does not work well in GraphQL applications.
+ * By default, Nest GraphQL applications use the ExternalExceptionFilter, which just rethrows the error:
+ * https://github.com/nestjs/nest/blob/master/packages/core/exceptions/external-exception-filter.ts
+ *
+ * The ExternalExceptinFilter is not exported, so we reimplement this filter here.
+ */
+class SentryGlobalGraphQLFilter {
+  public static readonly __SENTRY_INTERNAL__ = true;
+
+  /**
+   * Catches exceptions and reports them to Sentry.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  public catch(exception: unknown, host: ArgumentsHost): void {
+    captureException(exception);
+    throw exception;
```

Comment:
m: I am a bit sus because we identified previously for HTTP contexts taht throwing in error filters will not make the behaviour transparent. Is this the case for graphql?

Reply:
Did you check what the server returns before/after you added the filter? We need to be careful.

Reply:
Yes, responses are the same with/without filter. I also now added the logging logic from the `ExtenalExceptionFilter` so now output also aligns perfectly.